### PR TITLE
feat: implement list item component with partial functionality

### LINKED ISSUE #452: `Submission List Item Component` for Moderation Panel
# Problem
There is no component that can take in the data from the database that will display the data required

# Success
This component can take in props that are displayed as shown on [Figma](https://www.figma.com/file/QXQxlw4jNFIhMxkKU7bbwq/Find-a-doc%2C-Japan?type=design&node-id=1621-17300&mode=design&t=YlT64bYAuVCI537V-4 ).

Image

Image

- @ermish suggested that since we do not have access to the facility name at the moment that the facility name in this component should show the `submissionId`

Should be just a singular line of this whole image. That can be reused in a loop.

### Requirements

- [ ] The component should display a `number`. `status`, `modified`, `submitted`
- [ ] Matches Figma
- [ ] Accepts props to be used with provided data

# Implementation notes
Currently the backend does not provide a status. So this might have to be hardcoded when first accepting the props when data has been fetched

### PR DESCRIPTION
**Pull request by ShizuokaTerrier**
Resolves #452 

## 🔧 What changed
We created:

-  a component that displays the fake data in the ModSubmissionListContainer.
- an on-click event to take you to the relevant submission form. 
- In future we need to create a ticket to set up the logic for connecting specific data entries to specific submission forms.  

## 📸 Screenshots

-   ### Before
image

-   ### After
image

### CODE REVIEW

**Comment by theyokohamalife** [comment]
It's looking really good! A couple small suggestions about Git:
1. PR titles and commit messages generally don't end with a period. Please update it when you can.
2. We can see who the co-authors are in the commit history, so it's okay to not include them in the description.

**Comment by theyokohamalife** [comment]
If you want to follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines for your commit message, I'd suggest rewording this as:
`feat: implement list item component with partial functionality`

**Comment by NabbeunNabi** [comment]
updated based on feedback

**Comment by NabbeunNabi** [comment]
@theyokohamalife Have you had a chance to look at these changes?

**Review by theyokohamalife — APPROVED** [review]
Looks like good progress towards getting the feature working!
